libraries/AP_Math/fourier.cpp

```diff
@@ -0,0 +1,219 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+ 
+ //designed by Philippe Crochat / Anemos Technologies : pcrochat@anemos-technologies.com
+
+#include "fourier.h"
+
+Fourier_Analysis::Fourier_Analysis(){
+	_result_filter.set_cutoff_frequency(RESULT_FILTER_FREQ);
+}
+
+void Fourier_Analysis::Set_Fourier_Analysis(int32_t buffer_size){
```

Comment:
since it gets complicated to handle the case where buffer_size changes at runtime, why not have it initialized in the constructor then move this function into the Fourier_Analysis constructor like you did with the cutoff_freq.